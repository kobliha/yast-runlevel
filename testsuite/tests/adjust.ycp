/**
 * YaST2 test case for runlevel editor
 *
 * tests module adjust_runlevel
 */
{
    include "testsuite.ycp";

    map READ = $[];
    map WRITE = $[];
    map EXEC = $[];

    // wrong arguments
    TEST (`runlevel_adjust ("lpd"), [READ, WRITE, EXEC], false);
    // lpd script does not exist
    TEST (`runlevel_adjust ("lpd", "enable"), [READ, WRITE, EXEC], false);

    //
    // TEST ENABLE
    //
    DUMP ("--- Test enable 1 ---");
    
    // Disabled by START_ variable
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : ["4", "5", ],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ]
            ],
        "rc" : $[
            "system" : "no",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "enable"), [READ, WRITE, EXEC], false);

    DUMP ("--- Test enable 2 ---");
    // Disabled by links
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : [],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ],
            ],
        "rc" : $[
            "system" : "yes",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "enable"), [READ, WRITE, EXEC], false);

    DUMP ("--- Test enable 3 ---");
    // already enabled
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : [ "3", "4", ],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ],
            ],
        "rc" : $[
            "system" : "yes",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "enable"), [READ, WRITE, EXEC], false);
    
    //
    // TEST DISABLE
    //
    DUMP ("--- Test disable 1 ---");
    
    // Disabled by START_ variable
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : ["4", "5", ],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ]
            ],
        "rc" : $[
            "system" : "no",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "disable"), [READ, WRITE, EXEC], false);

    DUMP ("--- Test disable 2 ---");
    // Disabled by links
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : [],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ],
            ],
        "rc" : $[
            "system" : "yes",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "disable"), [READ, WRITE, EXEC], false);

    DUMP ("--- Test disable 3 ---");
    // already enabled
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : [ "3", "4", ],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ],
            ],
        "rc" : $[
            "system" : "yes",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "disable"), [READ, WRITE, EXEC], false);
    
    //
    // TEST SET DEFAULT
    //
    DUMP ("--- Test set default 1 ---");
    // Disabled by START_ variable
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : ["4", "5", ],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ]
            ],
        "rc" : $[
            "system" : "no",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "default"), [READ, WRITE, EXEC], false);

    DUMP ("--- Test set default 2 ---");
    // Disabled by links
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : [],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ],
            ],
        "rc" : $[
            "system" : "yes",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "default"), [READ, WRITE, EXEC], false);

    DUMP ("--- Test set default 3 ---");
    // already enabled
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : [ "3", "4", ],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ],
            ],
        "rc" : $[
            "system" : "yes",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "default"), [READ, WRITE, EXEC], false);
    
    DUMP ("--- Test set default 4 ---");
    // already enabled and already default
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : [ "1", "2", ],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ],
            ],
        "rc" : $[
            "system" : "yes",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "default"), [READ, WRITE, EXEC], false);

    DUMP ("--- Test set start ---");
    // start/stop/restart is always the same
    TEST (`runlevel_adjust ("lpd", "start"), [READ, WRITE, EXEC], false);

    DUMP ("--- Test set enable and start ---");
    // Disabled by START_ variable
    READ = $[
        "init" : $[
            "scripts" : $[
                "exists" : true,
                "runlevel" : $[
                    "lpd" : $[
                        "start" : ["4", "5", ],
                        "stop" : [],
                        ],
                    ],
                "comment" : $[
                    "lpd" : $[
                        "defstart" : [ "1", "2",],
                        "defstop" : ["3", "4", ],
                        "description" : "some description",
                        ],
                    ],
                ]
            ],
        "rc" : $[
            "system" : "no",
            ],
        ];
    // lpd script exist
    TEST (`runlevel_adjust ("lpd", "enable", "start"), [READ, WRITE, EXEC], false);
    
}
