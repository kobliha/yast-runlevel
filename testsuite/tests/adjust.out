Log	Usage: CallModule ("runlevel_adjust", [ "servicename", "action"[, "action", ...] ])
Return	-1
Read	.init.scripts.exists "lpd" false
Return	-1
Dump	--- Test enable 1 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":["4", "5"], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "no"
Write	.rc.system."START_LPD" "yes" true
Execute	.target.bash "/sbin/insserv lpd,start=1,2" 0
Return	`ok
Dump	--- Test enable 2 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":[], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "yes"
Execute	.target.bash "/sbin/insserv lpd,start=1,2" 0
Return	`ok
Dump	--- Test enable 3 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":["3", "4"], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "yes"
Return	`ok
Dump	--- Test disable 1 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":["4", "5"], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "no"
Return	`ok
Dump	--- Test disable 2 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":[], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "yes"
Return	`ok
Dump	--- Test disable 3 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":["3", "4"], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "yes"
Execute	.target.bash "/sbin/insserv lpd,start=" 0
Return	`ok
Dump	--- Test set default 1 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":["4", "5"], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "no"
Write	.rc.system."START_LPD" "yes" true
Execute	.target.bash "/sbin/insserv lpd,start=1,2" 0
Return	`ok
Dump	--- Test set default 2 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":[], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "yes"
Execute	.target.bash "/sbin/insserv lpd,start=1,2" 0
Return	`ok
Dump	--- Test set default 3 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":["3", "4"], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "yes"
Execute	.target.bash "/sbin/insserv lpd,start=1,2" 0
Return	`ok
Dump	--- Test set default 4 ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":["1", "2"], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "yes"
Return	`ok
Dump	--- Test set start ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":["1", "2"], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "yes"
Execute	.target.bash "TERM=raw /etc/init.d/lpd start" 0
Return	`ok
Dump	--- Test set enable and start ---
Read	.init.scripts.exists "lpd" true
Read	.init.scripts.runlevel "lpd" $["lpd":$["start":["4", "5"], "stop":[]]]
Read	.init.scripts.comment "lpd" $["lpd":$["defstart":["1", "2"], "defstop":["3", "4"], "description":"some description"]]
Read	.rc.system."START_LPD" "no"
Write	.rc.system."START_LPD" "yes" true
Execute	.target.bash "/sbin/insserv lpd,start=1,2" 0
Execute	.target.bash "TERM=raw /etc/init.d/lpd start" 0
Return	`ok
